From cfa5e2a8dab746c14c2e5876a48ff0d66cdf7486 Mon Sep 17 00:00:00 2001
From: Don Olmstead <don.olmstead@sony.com>
Date: Fri, 31 Mar 2023 19:46:07 -0700
Subject: [PATCH] Cherry-pick 262461@main (b36decf27ea9).
 https://bugs.webkit.org/show_bug.cgi?id=254852

    Fix !ENABLE(WEB_AUDIO) build after 262451@main
    https://bugs.webkit.org/show_bug.cgi?id=254852

    Unreviewed build fix.

    Add !ENABLE(WEB_AUDIO) guard.

    * Source/WebCore/page/MemoryRelease.cpp:

    Canonical link: https://commits.webkit.org/262461@main
---
 Source/WebCore/page/MemoryRelease.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Source/WebCore/page/MemoryRelease.cpp b/Source/WebCore/page/MemoryRelease.cpp
index d2d5009785bc..aa0c8cff9571 100644
--- a/Source/WebCore/page/MemoryRelease.cpp
+++ b/Source/WebCore/page/MemoryRelease.cpp
@@ -103,7 +103,9 @@ static void releaseCriticalMemory(Synchronous synchronous, MaintainBackForwardCa
     }
 
     CSSValuePool::singleton().drain();
+#if ENABLE(WEB_AUDIO)
     HRTFElevation::clearCache();
+#endif
 
     Page::forEachPage([](auto& page) {
         page.cookieJar().clearCache();
-- 
2.39.2

